src/core/models/calculateAggregateMetrics.ts

```diff
@@ -47,6 +48,40 @@ interface AggregateMetricsInterface<T> {
   records: T[] | undefined;
 }
 
+export function extrapolateRecordsForRange<T extends TimeSeriesDataRecord>(
+  metric: AggregateMetricsInterface<T>,
+  monthRange: number,
+  defaults: T
+): T[] {
+  const { mostRecentDate, records } = metric;
+  const earliestDate = startOfMonth(subMonths(mostRecentDate, monthRange));
+  const grouped = groupBy(properties("year", "month"))(records);
+  const allDates: T[] = [];
+
+  // For each month in range, if we have an entry for it add it to the array. If not, add an entry
+  // with a count of 0.
+  eachMonthOfInterval({ start: earliestDate, end: mostRecentDate }).forEach(
+    (date) => {
+      const found = grouped[`${date.getFullYear()},${date.getMonth() + 1}`];
+      const val =
+        found && found[0]
+          ? found[0]
+          : {
+              year: date.getFullYear(),
+              month: date.getMonth() + 1,
+              count: 0,
+            };
```

Comment:
You can use `?.[0]` as a null-safe accessor on `found`. This then allows you to use the null-coalescing operator (`??`) instead of a ternary operator.

```suggestion
      const val =
        found?.[0]
          ?? {
              year: date.getFullYear(),
              month: date.getMonth() + 1,
              count: 0,
            };
```

Reply:
👍 thanks!